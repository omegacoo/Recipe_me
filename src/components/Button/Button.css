.Button {
    display: block;
    border: 1px solid black;
    border-radius: 15px;
    width: 35vw;
    height: 20vh;
    text-align: center;
    padding-top: 7.5vh;
    margin-left: auto;
    margin-right: auto;
    margin-top: 5vh;
    text-decoration: none;
    font-size: 3vh;
}