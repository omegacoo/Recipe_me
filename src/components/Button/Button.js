import React from 'react';
import { Link } from 'react-router-dom';

import StoreContext from '../../StoreContext';

import './Button.css';

class Button extends React.Component {
    static contextType = StoreContext;

    render(){
        return(
            <Link 
                className='Button'
                to={this.props.target}
            >
                {this.props.name}
            </Link>
        );
    };
};

Button.defaultProps = {
    target: '/'
};

export default Button;