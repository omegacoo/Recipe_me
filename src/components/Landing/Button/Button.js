import React from 'react';
import { Link } from 'react-router-dom';

import StoreContext from '../../../StoreContext';

import './Button.css';

export default class Button extends React.Component {
    static contextType = StoreContext;

    render(){
        return(
            <Link 
                className='Button'
                to={this.props.name === 'guest' ? '/inputSelect' : `/${this.props.name}`}
            >
                {this.props.name}
            </Link>
        );
    };
};