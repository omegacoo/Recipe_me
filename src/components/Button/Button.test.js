import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Button from './Button';

describe('Button component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(
            <BrowserRouter>
                <Button />
            </BrowserRouter>, 
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders UI correctly', () => {
        expect(
        renderer.create(
            <BrowserRouter>
                <Button />
            </BrowserRouter>).toJSON()
        ).toMatchSnapshot();
    });
});