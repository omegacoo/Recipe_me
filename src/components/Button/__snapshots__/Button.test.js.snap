// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders UI correctly 1`] = `
<a
  className="Button"
  href="/"
  onClick={[Function]}
/>
`;
